drivers/include/sps30.h

```diff
@@ -0,0 +1,287 @@
+ *            may happen that the sensor is not yet ready to serve data.
+ *            Handling this within the driver simplifies application code by
+ *            omitting sleep handling or retries there.
+ *            This value may be overwritten to 1 if more fine-grained feedback
+ *            is required or even increased if the device is connected over
+ *            suboptimal wiring.
+ *
+ */
+#ifndef SPS30_ERROR_RETRY
+#define SPS30_ERROR_RETRY    (500U)
+#endif
+
+/**
+ * @brief       Initialize SPS30 sensor driver.
+ *              On success the measurement mode will be active after calling.
+ *
+ * @param[out]  dev         Pointer to an SPS30 device handle
+ * @param[in]   params      Parameters for device initialization
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_init(sps30_t *dev, const sps30_params_t *params);
+
+/**
+ * @brief       Enable the measurement action.
+ *
+ * @details     Starting the measurement activates the fan of the sensor and
+ *              increases the power consumption from below 8 mA to an average
+ *              of 60 mA. When the fan is starting up the consumption can reach
+ *              up to 80 mA for around 200 ms.
+ *              The measurement mode will stay active until either the power is
+ *              turned off, a stop is requested (@ref sps30_stop_measurement),
+ *              or the device is reset (#sps30_reset).
+ *
+ * @param[in]   dev               Pointer to an SPS30 device handle
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_start_measurement(const sps30_t *dev);
+
+/**
+ * @brief       Stops the measurement action.
+ *
+ * @details     Stopping the measurement sets the device back to idle mode.
+ *
+ * @param[in]   dev               Pointer to an SPS30 device handle
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_stop_measurement(const sps30_t *dev);
+
+/**
+ * @brief       Ask the device if a measurement is ready for reading.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ * @param[out]  error      Pre-allocated memory to return #sps30_error_code_t
+ *                         or NULL if not interested
+ *
+ * @return      true       if a new measurement is available
+ * @return      false      if no new measurement is available
+ */
+bool sps30_data_ready(const sps30_t *dev, int *error);
+
+/**
+ * @brief       Read a set of particulate matter measurements.
+ *
+ * @param[in]   dev        Pointer to an SPS30 device handle
+ * @param[out]  data       Pre-allocated memory to return measurements
+ *
+ * @return      #SPS30_OK on success, negative #sps30_error_code_t on error
+ */
+int sps30_read_measurement(const sps30_t *dev, sps30_data_t *data);
+
+/**
+ * @brief       Read the fan auto-clean interval.
+ *
+ * @details     The default value is 604800 seconds (1 week).
```

Comment:
IIRC the usage of `@details` is  superfluous, and can be omitted - but that's just cosmetic.

Reply:
I like the added information so I ignored this one ;)